src/cephadm/cephadmlib/node_proxy/redfish_client.py

```diff
@@ -0,0 +1,112 @@
+import json
+from urllib.error import HTTPError, URLError
+from .baseclient import BaseClient
+from .util import Logger, http_query
+from typing import Dict, Any, Tuple, Optional
+from http.client import HTTPMessage
+
+
+class RedFishClient(BaseClient):
+    PREFIX = '/redfish/v1/'
+
+    def __init__(self,
+                 host: str = '',
+                 port: int = 443,
+                 username: str = '',
+                 password: str = ''):
+        super().__init__(host, username, password)
+        self.log: Logger = Logger(__name__)
+        self.log.logger.info(f'Initializing redfish client {__name__}')
+        self.host: str = host
+        self.port: int = port
+        self.url: str = f'https://{self.host}:{self.port}'
+        self.token: str = ''
+        self.location: str = ''
+
+    def login(self) -> None:
+        if not self.is_logged():
+            self.log.logger.info('Logging in to '
+                                 f"{self.url} as '{self.username}'")
+            oob_credentials = json.dumps({'UserName': self.username,
+                                          'Password': self.password})
+            headers = {'Content-Type': 'application/json'}
+
+            try:
+                _headers, _data, _status_code = self.query(data=oob_credentials,
+                                                           headers=headers,
+                                                           endpoint='/redfish/v1/SessionService/Sessions/')
+                if _status_code != 201:
+                    self.log.logger.error(f"Can't log in to {self.url} as '{self.username}': {_status_code}")
+                    raise RuntimeError
+            except URLError as e:
+                msg = f"Can't log in to {self.url} as '{self.username}': {e}"
+                self.log.logger.error(msg)
+                raise RuntimeError
+            self.token = _headers['X-Auth-Token']
+            self.location = _headers['Location']
+
+    def is_logged(self) -> bool:
```

Comment:
can this be renamed to something like `is_logged_in`? `is_logged` makes me think it's checking if some log message was actually logged or not.